cognite_toolkit/_cdf_tk/commands/dump_data.py

```diff
@@ -290,6 +299,51 @@ def _get_resource_columns(self) -> list[SchemaColumn]:
         return columns
 
 
+class EventFinder(AssetCentricFinder[Event]):
+    supported_formats = frozenset({"csv", "parquet"})
+
+    def _create_loader(self, client: ToolkitClient) -> ResourceLoader:
+        return EventLoader.create_loader(client)
+
+    def _aggregate_count(self, hierarchies: list[str], data_sets: list[str]) -> int:
+        return self.client.events.aggregate_count(
+            filter=EventFilter(
+                data_set_ids=[{"externalId": item} for item in data_sets] or None,
+                asset_subtree_ids=[{"externalId": item} for item in hierarchies] or None,
+            )
+        )
+
+    def _get_resource_columns(self) -> list[SchemaColumn]:
+        columns = [
+            SchemaColumn(name="externalId", type="string"),
+            SchemaColumn(name="dataSetExternalId", type="string"),
+            SchemaColumn(name="startTime", type="integer"),
+            SchemaColumn(name="endTime", type="integer"),
+            SchemaColumn(name="type", type="string"),
+            SchemaColumn(name="subtype", type="string"),
+            SchemaColumn(name="description", type="string"),
+            SchemaColumn(name="assetExternalIds", type="string", is_array=True),
+            SchemaColumn(name="source", type="string"),
+        ]
+        data_set_ids = self.client.lookup.data_sets.id(self.data_sets) if self.data_sets else []
+        root_ids = self.client.lookup.assets.id(self.hierarchies) if self.hierarchies else []
+        metadata_keys = metadata_key_counts(self.client, "events", data_set_ids or None, root_ids or None)
```

Comment:
This is a pretty hefty way to order your keys. Is doing it this way really necessary? I realize this is already an established pattern, but maybe explain why? The key counts can change, so the ordering could change between dumps, and running a transformation to count keys is a pretty heavy method. Is it about knowing all the columns before-hand? This is running a preview so I don't think that will actually work in every case, since previews only run on a subset of the data (?). You'd need to run a full transformation.

Reply:
Unfortunately, yes. The aggregate endpoint for files (and all other asset-centric resources) lower cases the metadata keys. This is not what our users expects, as there is typically good information in the casing of metadata. Ref https://cognitedata.slack.com/archives/C019GLXD0T0/p1736754054881239

Yes this is for knowing all the columns before writing. The way preview is used here it actually runs on all rows, see the specified limits.

Reply:
Huh. I was not aware that previews actually had no upper limit. Tested it myself now and it seems you are right. I understand, this seems to be the most reasonable solution for this, though it feels fairly heavy.